testing_utils_test.go

```diff
@@ -165,6 +152,52 @@ func newFakeMarathonEndpoint(t *testing.T, configs *ConfigContainer) *endpoint {
 	}
 }
 
+// basicAuthMiddleware handles basic auth
+func basicAuthMiddleware(config *configContainer, next http.HandlerFunc) func(http.ResponseWriter, *http.Request) {
```

Comment:
It should suffice to pass the username and password parameters instead of the entire configuration to this function as we don't need the rest.